nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -326,24 +326,24 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
     public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
         boolean isCustomQuery = CUSTOM_QUERY.getValue().equals(context.getProperty(QUERY_TYPE).getValue());
 
+        FlowFile flowFile = session.get();
```

Comment:
I think the null check for the FlowFile would be still needed here, so it would be better to move that check condition outside from the `isCustomQuery` check.

Reply:
Thanks for the review, since we do want to evaluate EL wether the FlowFile is null or not, I added a null check in the ``processQuery`` part.

Reply:
I recommend removing flowfile == null check before the condition. Based @markap14 answer in case of processors annotated with @TriggerSerially this check is not necessary. However if a processor can have multiple executing thread, the following can happen:

"- 1 FlowFile in queue and 2 threads start running.
-  Thread 1 calls session.get() and gets the only FlowFile.
-  Then Thread 2 calls session.get() and gets back a null."

The flowfile is optional and the null check will be handled in the `processQuery` and `processCustomQuery` methods.